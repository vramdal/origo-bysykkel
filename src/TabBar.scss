.tab-bar {
  button:disabled {
    background-color: ButtonFace;
    color: ButtonText;
  }
  button:not(:disabled) {
    background-color: ButtonHighlight;
    color: ButtonText;
  }
}