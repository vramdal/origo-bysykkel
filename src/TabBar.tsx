import React from 'react';
import './TabBar.scss';

interface TabBarProps {
  activeId: string;
  switchTab: (tabId: string) => void;
  tabs: Tabs;
}

export type Tabs = Map<string, React.ReactNode>;

export function TabBar<T>({ activeId, switchTab, tabs }: TabBarProps): JSX.Element {
  return (
    <div className={'tab-bar'}>
      {Array.from(tabs.keys()).map((tabId: string) => {
        return (
          <button
            role={'tab'}
            key={tabId}
            disabled={tabId === activeId}
            type={'button'}
            onClick={(): void => switchTab(tabId)}
          >
            {tabs.get(tabId)}
          </button>
        );
      })}
    </div>
  );
}
