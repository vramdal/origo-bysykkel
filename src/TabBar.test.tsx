import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { TabBar, Tabs } from './TabBar';

const tabs: Tabs = new Map([
  ['tab1', 'Tab 1'],
  ['tab2', 'Tab 2'],
]);

describe('TabBar', () => {
  it('should render like the snapshot', () => {
    const { container } = render(<TabBar activeId={'tab1'} switchTab={jest.fn()} tabs={tabs} />);

    expect(container).toMatchSnapshot();
  });

  it('it should call switchTab when a non-disabled button is clicked', () => {
    const switchTab = jest.fn();

    const { getAllByRole } = render(<TabBar activeId={'tab1'} switchTab={switchTab} tabs={tabs} />);
    const buttons = getAllByRole('tab');
    fireEvent.click(buttons[1]);

    expect(switchTab).toHaveBeenCalledWith('tab2');
  });

  it('it should not call switchTab when a disabled button is clicked', () => {
    const switchTab = jest.fn();

    const { getAllByRole } = render(<TabBar activeId={'tab1'} switchTab={switchTab} tabs={tabs} />);
    const buttons = getAllByRole('tab');
    fireEvent.click(buttons[0]);

    expect(switchTab).not.toHaveBeenCalled();
  });
});
