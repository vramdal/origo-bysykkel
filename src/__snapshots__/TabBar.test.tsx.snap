// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBar should render like the snapshot 1`] = `
<div>
  <div
    class="tab-bar"
  >
    <button
      disabled=""
      role="tab"
      type="button"
    >
      Tab 1
    </button>
    <button
      role="tab"
      type="button"
    >
      Tab 2
    </button>
  </div>
</div>
`;
